### PR TITLE
CON-2931 Resolve missing assets for `stealth-boom` project

### LINKED ISSUE #2566: [BUG] Issue in the `stealth-boom` gaming project
**Describe the bug**

In the task description it says: 

"The basics assets you will need for this project can be found in the [Stealth-Boom.zip](https://assets.01-edu.org/Unreal-Engine-Projects/StealthBoom.zip) file. It contains the basic animations, character, enemies and props you will need for this project."

After downloading the StealthBoom.zip it only contains Drone.fbx and the task description in a docx file.

So in other words the file provided to us students is missing the basic animations, character, enemies and props. 

**Users**

Students following the Gaming specialization, grit:lab, Åland.

**Severity**

(‼️ major)

**Type**

(⚙️ functionality), (🗂️ documentation)

**To Reproduce**

Downloading the file

**Workarounds**

It would be helpful is you could provide the files as mentioned in the task description.

**Expected behavior**

The zip file should contain what is said in the task description. 

**Attachments**

N/A

**Desktop (please complete the following information):**

OS: mac OS, Sonoma
Browser: Google Chrome
Unreal Engine : 5.3.2
Epic Game Launcher: 16.1.1

**Smartphone (please complete the following information):**

N/A

**Additional context**
N/A

### LINKED ISSUE #2644: [DUPLICATE][BUG]  Issue `stealth-boom` subject
stealth-boom
---
The provided zip file does not contain necessary assets.

image

image

### PR DESCRIPTION
**Pull request by EldritchGriffin**
CON-2931

Refined the subject and audit of the stealth-boom project and provided a file that contains the basic assets required for the project

- Closes #2644 
- Closes #2566

### CODE REVIEW

File: subjects/gaming/stealth-boom/audit/README.md [hunk 1]
```diff
@@ -1,20 +1,10 @@
...
-###### Is there a sound for the player shooting?
-
-###### Are there bullets impacts present when shooting at a wall?
-
-###### When the player is hit, are there any blood particles?
-
-###### Is a main menu with the five options displayed on the screen?
+###### Is the main menu with the five options displayed on the screen?
 
 ###### Can the general sound of the game be managed directly on the settings menu?
```
Comment (MSilva95) [inline subjects/gaming/stealth-boom/audit/README.md:10]: `When changing the resolution, does a confirmation pop-up appear in the center of the screen asking if the player wants to keep the new graphics settings?`

`If the player presses 'No' on the graphics confirmation pop-up, or if the pop-up is not confirmed within 10 seconds, do the settings revert to the previous ones?`

File: subjects/gaming/stealth-boom/README.md [hunk 2]
```diff
@@ -1,82 +1,98 @@
...
+  - Pickable Health around the map.
+
+- Player:
 
+  - Walk and run animations.
+  - Shoot animation.
+  - Duck animation.
+  - Melee attack animation.
+  - Blood particles when hit.
+  - A health bar that should go down whenever the player gets damaged. When the player dies, he has the choice to either quit the game, go back to the main menu or start over.
```
Comment (MSilva95) [inline subjects/gaming/stealth-boom/README.md:58]: this is a new topic should be separated 
`- When the player dies, he has the choice to either quit the game, go back to the main menu or start over.`

File: subjects/gaming/stealth-boom/README.md [hunk 3]
```diff
@@ -1,82 +1,98 @@
...
+  - The player mission is up to you, it can be some task to fix, kill all enemies without getting caught or collect documents... Whatever you choose, the player should have enemies on his way to divert him away from his objective.
   - When the player successfully completes his mission, a pop up should appear saying that the mission is completed.
 
-- The player has a health bar that should go down whenever the player gets shot. When the player dies, he has the choice to either quit the game, go back to the main menu or start over.
+- Gun mechanics:
+
```
Comment (MSilva95) [inline subjects/gaming/stealth-boom/README.md:63]: we should add Shooting here as well

File: subjects/gaming/stealth-boom/audit/README.md [hunk 4]
```diff
@@ -24,46 +14,70 @@
...
 
-###### Can the camera close some part of the map (thru closed doors, open traps and tries to kill the player etc…) to the player when the player is being detected?
+###### Does the player have a health bar?
 
-###### Are Behavior Trees used to implement the AI of the Guards and Drones?
+###### When the player loses all his health (dies), does he get to choose whether to quit the game, go back to the main menu, or to start over?
 
 ###### Does the player have a goal?
 
 ###### When the goal of the player is successfully completed, does a pop up appear saying that the mission is completed?
```
Comment (MSilva95) [inline subjects/gaming/stealth-boom/audit/README.md:46]: Separate the `Gun mechanics:`, organize questions and add missing question here is a suggestion:

```
##### Player Mechanics

###### Does the player have all the minimal animation required (walking, running, melee attacking, shooting, ducking)?

###### Does the player have a health bar?

###### Does the player health decreases when he gets damaged?

###### When the player is hit, are there any blood particles?

###### When the player loses all his health (dies), does he get to choose whether to `quit the game`, go back to the `main menu`, or to `start over`?

###### Does the player have a defined goal or mission?

###### Whatever the goal or mission, can you confirm that the player had enemies on his way to divert him away from his objective?

###### When the player successfully completes the goal or mission, does a pop-up appear indicating `mission completion`?

#### Gun Mechanics

###### Is there a sound for the guns shooting?

###### Is there a widget displaying the amount of bullets the player has?

###### Is the player unable to shoot when he has no bullets?

###### Are there bullets impacts present when shooting at a wall?

```

File: subjects/gaming/stealth-boom/README.md [hunk 5]
```diff
@@ -1,82 +1,98 @@
...
+  - a widget displaying the number of bullets.
+  - when the number of bullets is 0 the player is unable to shoot.
+  - shooting should have a sound effect and visual effect.
+  - Bullets visual impact on walls.
```
Comment (MSilva95) [inline subjects/gaming/stealth-boom/README.md:67]:
```
  - A widget should display the current number of bullets available to the player.
  - The player should be unable to shoot when the bullet count reaches zero.
  - Shooting should include both a sound effect and a visual effect.
  - Bullet impacts on walls should be visually represented.
```

File: subjects/gaming/stealth-boom/README.md [hunk 6]
```diff
@@ -1,82 +1,98 @@
...
+
+- Enemy:
+
+  - Your game should contain atleast 2 enemy types `Melee` and `Ranged`.
```
Comment (MSilva95) [inline subjects/gaming/stealth-boom/README.md:71]: Typo `at least`

File: subjects/gaming/stealth-boom/audit/README.md [hunk 7]
```diff
@@ -24,46 +14,70 @@
...
-###### Does the player health decreases when he gets shot by the guards?
+###### Does the game contain at least 2 enemy types (ranged and melee)
 
-###### When the player loses all his health (dies), does he get to choose whether to quit the game, go back to the main menu, or to start over?
+###### Are Behavior Trees used to implement the AI of the enemies?
 
-###### If the player starts over does he spawn back at the starting point?
+###### Do the enemies wander around the map?
+
+###### When a player enters the field of view of an enemy, does it switch to chasing mode
```
Comment (MSilva95) [inline subjects/gaming/stealth-boom/audit/README.md:55]:
```
###### Does the game include at least two types of enemies: `Melee` and `Ranged`?
###### Is enemy AI implemented using Behavior Trees?
###### Do the enemies wander around the map?
###### When a player enters an enemy's field of view, does the enemy switch to a chasing mode?
```

File: subjects/gaming/stealth-boom/audit/README.md [hunk 8]
```diff
@@ -24,46 +14,70 @@
...
+
+###### Do melee enemies approach the player to perform melee attacks?
+
+###### Do ranged enemies take cover and shoot at the player?
+
+###### do the enemies go back to patrol mode when the player is hidden from all field of views for a set amount of time?
```
Comment (MSilva95) [inline subjects/gaming/stealth-boom/audit/README.md:61]: Lowercase at start of the question, must be upper

File: subjects/gaming/stealth-boom/audit/README.md [hunk 9]
```diff
@@ -24,46 +14,70 @@
...
+
+###### do the enemies have a sound effect and visual effect when entering and exiting chase mode?
```
Comment (MSilva95) [inline subjects/gaming/stealth-boom/audit/README.md:63]: Lowercase at start of the question, must be upper

File: subjects/gaming/stealth-boom/audit/README.md [hunk 10]
```diff
@@ -24,46 +14,70 @@
...
+
+###### are enemies in chase mode alerting nearby enemies ?
```
Comment (MSilva95) [inline subjects/gaming/stealth-boom/audit/README.md:65]: Lowercase at start of the question, must be upper

File: subjects/gaming/stealth-boom/README.md [hunk 11]
```diff
@@ -1,82 +1,98 @@
...
+  - enemy AI should be made with behavior trees.
+  - enemy should be able to patrol around the map in a predefined path.
+  - enemy detects the player, if the player crosses its field of view.
```
Comment (MSilva95) [inline subjects/gaming/stealth-boom/README.md:74]:
```
  - The game should feature at least two types of enemies: Melee and Ranged.
  - Behavior trees should be used to implement enemy AI.
  - Enemies should be able to patrol predefined paths around the map.
  - Enemies should detect the player if the player enters their field of view.
   ```

File: subjects/gaming/stealth-boom/README.md [hunk 12]
```diff
@@ -1,82 +1,98 @@
...
+  - When the player enters the field of view of an enemy, the enemy enters into chasing mode and must start running after the player.
+    - `Ranged` enemies should take cover and shoot at the player.
+    - `melee` enemies should run close to the player and perform melee attacks.
+  - Enemies in chase mode alert nearby enemies making them enter chase mode as well.
+  - If the player leaves the field of view of all enemies for a certain time, the enemies go back to patrol mode.
+  - Enemies should have sounds effect whenever they change from chase to patrol mode, as well as the other way around.
+  - Enemies should have a visual indicator showing whether they are in patrol or chase mode
+
```
Comment (MSilva95) [inline subjects/gaming/stealth-boom/README.md:82]: Maybe add this header here as well  as we have it on the audit
" Game loop"

File: subjects/gaming/stealth-boom/README.md [hunk 13]
```diff
@@ -1,82 +1,98 @@
...
+- When pressing `Esc` the game is set on paused and a widget similar to the main menu widget pops up.
+- The player should be able to change the game graphics setting exactly like in the main menu.
+- A game should not last longer than 6 minutes. After that the same choices should appear as when the player dies.
```
Comment (MSilva95) [inline subjects/gaming/stealth-boom/README.md:85]: This contradicts the question in the audit:

`###### Is the lifespan of the game at least 6 minutes long from launch to mission completed?`

File: subjects/gaming/stealth-boom/audit/README.md [hunk 14]
```diff
@@ -24,46 +14,70 @@
...
+
+##### Game loop
+
+###### When 'esc' is pressed does the game pause and a widget similar to the main menu is displayed?
+
+###### can the player perform all the actions that appear in the menu when the game is paused?
```
Comment (MSilva95) [inline subjects/gaming/stealth-boom/audit/README.md:71]: Lowercase here as well

File: subjects/gaming/stealth-boom/audit/README.md [hunk 15]
```diff
@@ -24,46 +14,70 @@
...
+
+###### do the enemies go back to patrol mode when the player is hidden from all field of views for a set amount of time?
+
+###### do the enemies have a sound effect and visual effect when entering and exiting chase mode?
+
+###### are enemies in chase mode alerting nearby enemies ?
+
+##### Game loop
+
+###### When 'esc' is pressed does the game pause and a widget similar to the main menu is displayed?
```
Comment (MSilva95) [inline subjects/gaming/stealth-boom/audit/README.md:69]: instead of 'esc' use``` `Esc` ```

File: subjects/gaming/stealth-boom/README.md [hunk 16]
```diff
@@ -1,82 +1,98 @@
...
 
-  - If the player starts over, the level should not be reloaded. The player should spawn back to the starting point.
+### Bonus
```
Comment (MSilva95) [inline subjects/gaming/stealth-boom/README.md:87]: Capitalize the first letter of each sentence of the bonus section

File: subjects/gaming/stealth-boom/audit/README.md [hunk 17]
```diff
@@ -24,46 +14,70 @@
...
 ###### Is the lifespan of the game at least 6 minutes long from launch to mission completed?
 
 #### Bonus
 
-###### +Are there headshots implemented?
+###### +Did the student use different assets than the ones provided in the subject?
+
+###### +Are there more enemy types than the basic melee and ranged enemies?
+
+###### +Are there areas in the game that require an item to access ?
```
Comment (MSilva95) [inline subjects/gaming/stealth-boom/audit/README.md:81]: Extra space at near ?

**Review by MSilva95 — CHANGES_REQUESTED** [review]
Overall it's good, I made some suggestions and there were some things missing, some contradictions, but mostly it's grammar suggestions and improve questions order.

File: subjects/gaming/stealth-boom/README.md [hunk 18]
```diff
@@ -1,82 +1,98 @@
...
-  - Crouch animation
-    - the player should be able to do the above six actions while crouching
-  - Melee attack animation
-  - Gun sound when firing
-  - Bullets visual impact on walls (see decals documentation)
-  - Blood particles when hit
+The following aspects should be fulfilled:
 
-- The game should contain a main menu where the player can:
+- Main Menu:
```
[the remark this answers is not in the document]
Comment (MSilva95) [reply subjects/gaming/stealth-boom/README.md:36]: this is for all topics

[the remark this answers is not in the document]
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/README.md]: done

File: subjects/gaming/stealth-boom/README.md (hunk 2 above)
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/README.md]: done

File: subjects/gaming/stealth-boom/README.md (hunk 3 above)
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/README.md]: true! done

File: subjects/gaming/stealth-boom/README.md (hunk 11 above)
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/README.md]: done

File: subjects/gaming/stealth-boom/README.md (hunk 12 above)
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/README.md]: added

File: subjects/gaming/stealth-boom/README.md (hunk 16 above)
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/README.md:88]: done

File: subjects/gaming/stealth-boom/audit/README.md [hunk 19]
```diff
@@ -1,20 +1,10 @@
 > Due to file size reason, the solution might be uploaded on GitHub instead of Gitea!
 
-#### Functional
+#### Mandatory
```
[the remark this answers is not in the document]
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/audit/README.md]: changed

File: subjects/gaming/stealth-boom/audit/README.md [hunk 20]
```diff
@@ -1,20 +1,10 @@
...
-###### Does the player have all the minimal animation required (walking, running, melee attacking, aiming, reloading, shooting, crouching, crouch walking, crouch aiming, crouch reloading, crouch shooting)?
-
-###### Is there a sound for the player shooting?
-
-###### Are there bullets impacts present when shooting at a wall?
-
-###### When the player is hit, are there any blood particles?
-
-###### Is a main menu with the five options displayed on the screen?
+###### Is the main menu with the five options displayed on the screen?
```
[the remark this answers is not in the document]
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/audit/README.md]: i didnt change all sentences from the former dev, just rearranged and restructured and added my own.
from now on i will rephrase the whole subject when refining.
changed !

File: subjects/gaming/stealth-boom/audit/README.md [hunk 21]
```diff
@@ -1,20 +1,10 @@
...
 
 ###### Can the general sound of the game be managed directly on the settings menu?
```
[the remark this answers is not in the document]
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/audit/README.md]: changed

File: subjects/gaming/stealth-boom/audit/README.md (hunk 1 above)
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/audit/README.md:10]: changed

File: subjects/gaming/stealth-boom/audit/README.md (hunk 4 above)
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/audit/README.md]: seperated.

File: subjects/gaming/stealth-boom/audit/README.md (hunk 7 above)
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/audit/README.md]: changed

File: subjects/gaming/stealth-boom/audit/README.md (hunk 8 above)
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/audit/README.md]: changed

File: subjects/gaming/stealth-boom/audit/README.md (hunk 9 above)
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/audit/README.md]: changed

File: subjects/gaming/stealth-boom/audit/README.md (hunk 10 above)
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/audit/README.md]: changed

File: subjects/gaming/stealth-boom/audit/README.md (hunk 15 above)
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/audit/README.md]: changed

File: subjects/gaming/stealth-boom/audit/README.md (hunk 14 above)
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/audit/README.md]: changed

File: subjects/gaming/stealth-boom/audit/README.md (hunk 17 above)
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/audit/README.md]: removed

File: subjects/gaming/stealth-boom/README.md [hunk 22]
```diff
@@ -1,82 +1,99 @@
...
 ### Objectives
 
-The idea of this project is to create a little 10 minutes gameplay game with missions, with stealth based gameplay and with an AI patrolling NPC.
+The goal of this project is to create a playable gameplay loop around stealth mechanics.
+
+### Scenario
+
+You are an Unreal Engine developer tasked with creating a full stealth game from scratch. You need to implement everything, including AI behavior trees, player animations, and gun mechanics. As you progress, you realize balancing these elements—ensuring the AI responds accurately to player movements while maintaining smooth gameplay—is more challenging than expected. Every feature, from the main menu to mission completion, must work seamlessly to deliver a polished and playable game within the given constraints.
+
+Good luck and remember to have fun while making the game !
```
Comment (Oumaimafisaoui) [inline subjects/gaming/stealth-boom/README.md:17]: Minor: Adding comma after luck.

File: subjects/gaming/stealth-boom/README.md [hunk 23]
```diff
@@ -1,82 +1,99 @@
...
+
+- The basics assets you will need for this project can be found in the [StealthBoomAssets.zip](https://assets.01-edu.org/gamedev/stealth-boom-assets.zip) file. It contains the basic animations, character, enemies and props you will need for this project.
 
-The basics assets you will need for this project can be found in the [Stealth-Boom.zip](https://assets.01-edu.org/Unreal-Engine-Projects/StealthBoom.zip) file. It contains the basic animations, character, enemies and props you will need for this project.
+> NOTE: The assets in the file are intended to help streamline the process of locating assets, not to eliminate it.
+> TIP: Use [itch.io](https://www.itch.io) to get sound effects (not included in the assets file) or to find extra assets
 
 ### Instructions
 
 The following aspects should be fulfilled:
```
Comment (Oumaimafisaoui) [inline subjects/gaming/stealth-boom/README.md:34]: Minor: replace aspects with requirements.

File: subjects/gaming/stealth-boom/README.md [hunk 24]
```diff
@@ -1,82 +1,99 @@
...
-The idea of this project is to create a little 10 minutes gameplay game with missions, with stealth based gameplay and with an AI patrolling NPC.
+The goal of this project is to create a playable gameplay loop around stealth mechanics.
+
+### Scenario
+
+You are an Unreal Engine developer tasked with creating a full stealth game from scratch. You need to implement everything, including AI behavior trees, player animations, and gun mechanics. As you progress, you realize balancing these elements—ensuring the AI responds accurately to player movements while maintaining smooth gameplay—is more challenging than expected. Every feature, from the main menu to mission completion, must work seamlessly to deliver a polished and playable game within the given constraints.
+
+Good luck and remember to have fun while making the game !
+
+### Resources
```
Comment (Oumaimafisaoui) [inline subjects/gaming/stealth-boom/README.md:19]: Resources section can be put at the end of this subject.

File: subjects/gaming/stealth-boom/README.md [hunk 25]
```diff
@@ -1,82 +1,99 @@
...
-- When pressing `Esc` the game is set on paused and the main menu widget pops up.
+- Have more enemy types e.g a turret that is stationary but deals great damage, etc...
 
-  - The player should be able to change the game graphics setting exactly like in the main menu.
+- Have areas in the game that are locked behind doors that require keys which you can obtain from specific enemies
 
-- A game should not last longer than 6 minutes. After that the same choices should appear as when the player dies.
+- Have multiple different weapon types that you can pickup around the map and use to finish the mission
 
 > Do not forget to zip up the project compile and save everything for peer correction.
```
Comment (Oumaimafisaoui) [inline subjects/gaming/stealth-boom/README.md:98]: This can go under `submission` section.

File: subjects/gaming/stealth-boom/README.md (hunk 22 above)
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/README.md:17]: done

File: subjects/gaming/stealth-boom/README.md (hunk 23 above)
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/README.md:34]: done

File: subjects/gaming/stealth-boom/README.md (hunk 25 above)
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/README.md:98]: done

File: subjects/gaming/stealth-boom/README.md (hunk 24 above)
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/README.md:19]: done

File: subjects/gaming/stealth-boom/README.md (hunk 5 above)
Comment (MSilva95) [reply subjects/gaming/stealth-boom/README.md]: There is still a typo in this section, and I noticed you didn't change to the version I proposed as well, do you think it's better as it is?

Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/README.md]: i fixed the typo, i didn't keep the old version, i used your version to refine it.
mainly to keep the sentences concise and consistent with the rest of the subject.

Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/README.md]: i fixed the typo, i didn't keep the old version, i used your version to refine it.
mainly to keep the sentences concise and consistent with the rest of the subject.

File: subjects/gaming/stealth-boom/audit/README.md [hunk 26]
```diff
@@ -2,68 +2,90 @@
 
 #### Functional
 
-###### Does the map contain places for the player to hide from enemies?
+##### Main Menu
```
Comment (MSilva95) [inline subjects/gaming/stealth-boom/audit/README.md:5]: The title sections like :

```md
##### Main Menu
##### Enemies
##### Map / Level
...
```

they use 5 hashes, and this normally we only use to give students commands like this example:

```md
##### Check the Repo content

A `README.md` file and all files used to create, delete and manage the student infrastructure must be submitted in the repo.

###### Are all the required files present?
```

You can see we use 5 hashes for the command and 6 hashes for the actual yes/no question.
So in my opinion we should make the functional go away  and use 4 hashes for those topics or we can use ** 

```md
**Main Menu**
**Enemies**
**Map / Level**
...
```

@Oumaimafisaoui what do you think?

**Review by MSilva95 — CHANGES_REQUESTED** [review]

File: subjects/gaming/stealth-boom/audit/README.md (hunk 26 above)
Comment (EldritchGriffin) [reply subjects/gaming/stealth-boom/audit/README.md]: i removed functional and made all topics only 4 hashes

**Review by MSilva95 — APPROVED** [review]
LGTM!

**Pull request merged**